src/router/index.js

```diff
@@ -106,27 +33,30 @@ export function enableHistory(
   store,
   pushStateMountPoints = [],
   replaceStateMountPoints = [],
-  options
+  options = {}
 ) {
   const { debounceTime, useHash } = {
     ...defaultOptions,
-    ...(options || {})
+    ...options
   };
 
   store.syncToLocations = pushStateMountPoints.concat(replaceStateMountPoints);
   store.useHash = useHash;
 
-  const update = () => syncLocationToStore(store, pushStateMountPoints);
+  const update = mountpoints => {
+    store.syncedLocationToStore = true;
+    readUrl(store, mountpoints);
+    store.syncedLocationToStore = false;
+  };
 
-  window.addEventListener("popstate", update);
+  window.addEventListener("popstate", () => update(pushStateMountPoints));
```

Comment:
You might want to be able to tear down this listener; in that case, you should keep a reference to the function somewhere, else you'll be leaking refs in case the store gets recreated.